honeybee/model.py

```diff
@@ -119,243 +207,391 @@ def north_vector(self, value):
     @property
     def rooms(self):
         """A list of all Room objects in the model."""
-        return self._rooms
+        return tuple(self._rooms)
 
     @property
     def faces(self):
         """A list of all Face objects in the model."""
-        return self._faces
+        child_faces = []
+        for room in self._rooms:
+            child_faces.extend(room._faces)
+        return child_faces + self._faces
 
     @property
     def shades(self):
-        """A list of all Shades objects in the model."""
-        return self._shades
+        """A list of all Shade objects in the model."""
+        child_shades = []
+        for room in self._rooms:
+            child_shades.extend(room.shades)
+            for face in room.faces:
+                child_shades.extend(face.shades)
+                for ap in face._apertures:
+                    child_shades.extend(ap.shades)
+        for face in self._faces:
+            child_shades.extend(face.shades)
+            for ap in face._apertures:
+                child_shades.extend(ap.shades)
+        for ap in self._apertures:
+            child_shades.extend(ap.shades)
+        return child_shades + self._shades
 
     @property
     def apertures(self):
         """A list of all Aperture objects in the model."""
-        return self._apertures
+        child_apertures = []
+        for room in self._rooms:
+            for face in room._faces:
+                child_apertures.extend(face._apertures)
+        for face in self._faces:
+            child_apertures.extend(face._apertures)
+        return child_apertures + self._apertures
 
     @property
     def doors(self):
         """A list of all Door objects in the model."""
-        return self._doors
+        child_doors = []
+        for room in self._rooms:
+            for face in room._faces:
+                child_doors.extend(face._doors)
+        for face in self._faces:
+            child_doors.extend(face._doors)
+        return child_doors + self._doors
 
     @property
-    def has_ngon_apertures(self):
-        """Boolean noting whether the model has apertures with more than 4 sides.
-
-        This is helpful for energy simulation since EnergyPlus cannot accept
-        sub-faces with more than 4 sides. This property should be checked before
-        running the triangulate_ngon_apertures() method since many models do not
-        have such apertures.
-        """
-        for ap in self.apertures:
-            if len(ap.geometry) > 4:
-                return True
-        return False
+    def orphaned_faces(self):
+        """A list of all Face objects without parent Rooms in the model."""
+        return tuple(self._faces)
```

Comment:
I suggest to keep the naming as `orphaned_faces` and `_orphaned_faces`. I do not have good memories of breaking the rule in naming like this specially now that you have a `faces` property and naturally one would think `_face` corresponds to `faces` and not `orphaned_faces`.

Reply:
Same comment about all the other `orphaned_*` properties.

Reply:
Ok. That is a fair point.

Reply:
Changed!